internal/storage/clickhouse.go

```diff
@@ -0,0 +1,91 @@
+package storage
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type ClickHouseConnector struct {
+	cache clickhouse.Conn
+}
+
+type ClickhouseConnectorConfig struct {
+	ExpiresAt time.Duration
+}
+
+func NewClickHouseConnector(cfg *ClickhouseConnectorConfig) (*ClickHouseConnector, error) {
+	conn, err := ConnectDB()
+	// Question: Should we add the table setup here?
+	if err != nil {
+		return nil, err
+	}
+	return &ClickHouseConnector{
+		cache:  conn,
+	}, nil
+}
+
+
+func (c *ClickHouseConnector) Get(index, partitionKey, rangeKey string) (string, error) {
+	key := fmt.Sprintf("%s:%s", partitionKey, rangeKey)
+	// Does it make sense to check the expiration duration in the query?
+	query := "SELECT value FROM chainsaw.indexer_cache WHERE key = ?"
```

Comment:
What do you think about how the interface should look like?

`index, partitionKey, rangeKey string` probably don't suit us

We need to be able to define 
* a data type (to know if it's txs, blocks or logs and if it's staging or main table)
* a key to sort by (block number or tx hash?)

Reply:
These functions in the interface
`Get`
`GetMany`
`Insert` (current `Set`, but should support multiple objects)
`DeleteOne`
`DeleteMany`

Reply:
How about the Factory pattern as mentioned here: https://thirdwebdev.slack.com/archives/C0794ENCUE4/p1726606680097809